src/js/filterStar.js

```diff
@@ -0,0 +1,102 @@
+const starFrom = document.querySelector(".starsFrom a");
+const starsFrom = document.querySelectorAll(".starsFrom a");
+const starTo = document.querySelector(".starsTo a");
+const starsTo = document.querySelectorAll(".starsTo a");
+const ratingFil = document.querySelector(".ratingFilter");
+
+let activeIdx = null;
+
+starsFrom.forEach((starFrom, clickIdx) => {
+  starFrom.addEventListener("click", (e) => {
+    if (activeIdx === clickIdx) {
+      e.target.classList.remove("active");
+      activeIdx = clickIdx - 1;
+    } else {
+      e.target.classList.add("active");
+      activeIdx = clickIdx;
+    }
+
+    starsFrom.forEach((otherStar, otherIdx) => {
+      if (otherIdx < clickIdx) {
+        otherStar.classList.add("active");
+      } else if (otherIdx > clickIdx) {
+        otherStar.classList.remove("active");
+      }
+    });
+  });
+});
+
+starsTo.forEach((starTo, clickIdx) => {
+  starTo.addEventListener("click", (e) => {
+    if (activeIdx === clickIdx) {
+      e.target.classList.remove("active");
+      activeIdx = clickIdx - 1;
+    } else {
+      e.target.classList.add("active");
+      activeIdx = clickIdx;
+    }
+
+    starsTo.forEach((otherStar, otherIdx) => {
+      if (otherIdx < clickIdx) {
+        otherStar.classList.add("active");
+      } else if (otherIdx > clickIdx) {
+        otherStar.classList.remove("active");
+      }
+    });
+  });
+});
+
+var parentFrom = document.getElementById("starFrom");
+var valueFrom = parentFrom.getElementsByClassName("starDeactive active");
+console.log(valueFrom.length);
+
+var parentTo = document.getElementById("starTo");
+var valueTo = parentTo.getElementsByClassName("starDeactive active");
+var qtyClasses = valueTo.length;
+console.log(qtyClasses);
+
+ratingFil.addEventListener("click", (e) => {
+  if (e.target.tagName === "A") {
+    const rating = room.filter((character) => {
+      return (
+        character.rating >= valueFrom.length &&
+        character.rating <= valueTo.length
+      );
+    });
```

Comment:
Här är själva filtreringen? Tror den kan deklareras in i en separat funktion (utanför eventListener) med export så David kan få åtkomst till den!